domain/schema/model.go

```diff
@@ -44,3 +45,25 @@ CREATE TABLE model_config (
 );
 `)
 }
+
+func spacesSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE provider_spaces (
+    uuid            TEXT PRIMARY KEY,
+    name            TEXT
+);
+
+CREATE TABLE spaces (
+    uuid            TEXT PRIMARY KEY,
+    name            TEXT NOT NULL,
+    is_public       BOOLEAN,
+    provider_uuid   TEXT,
+    CONSTRAINT      fk_lease_pin_lease
+        FOREIGN KEY     (provider_uuid)
+        REFERENCES      provider_spaces(uuid)
+);
+
+CREATE UNIQUE INDEX idx_spaces_uuid_name
+ON spaces (uuid, name);
```

Comment:
This would make `uuid1, space1` and `uuid2, space1` valid tuples which is currently not the case in Mongo.

For now, the index should be on just `name`.